bundles/org.openhab.binding.rotel/src/main/java/org/openhab/binding/rotel/internal/protocol/hex/RotelHexProtocolHandler.java

```diff
@@ -0,0 +1,778 @@
+            logger.debug("Invalid check sum in response: {} rather than {}",
+                    Integer.toHexString(responseMessage[responseMessage.length - 1] & 0x000000FF),
+                    Integer.toHexString(checksum & 0x000000FF));
+            throw new RotelException("Invalid check sum in response");
+        }
+    }
+
+    /**
+     * Compute the checksum of a message
+     *
+     * @param message the buffer containing the message
+     * @param maxIdx the position in the buffer at which the sum has to be stopped
+     *
+     * @return the checksum as a byte
+     */
+    public static byte computeCheckSum(byte[] message, int maxIdx) {
+        int result = 0;
+        for (int i = 1; i <= maxIdx; i++) {
+            result += (message[i] & 0x000000FF);
+        }
+        return (byte) (result & 0x000000FF);
+    }
+
+    /**
+     * Analyze a valid HEX message and dispatch corresponding (key, value) to the event listeners
+     *
+     * @param incomingMessage the received message
+     */
+    @Override
+    protected void handleValidMessage(byte[] incomingMessage) {
+        if (incomingMessage[3] != STANDARD_RESPONSE) {
+            return;
+        }
+
+        final int idxChars = model.isCharsBeforeFlags() ? 4 : (4 + model.getRespNbFlags());
+
+        // Replace characters with code < 32 by a space before converting to a string
+        for (int i = idxChars; i < (idxChars + model.getRespNbChars()); i++) {
+            if (incomingMessage[i] < 0x20) {
+                incomingMessage[i] = 0x20;
+            }
+        }
+
+        String value = new String(incomingMessage, idxChars, model.getRespNbChars(), StandardCharsets.US_ASCII);
+        logger.debug("handleValidHexMessage: chars *{}*", value);
+
+        final int idxFlags = model.isCharsBeforeFlags() ? (4 + model.getRespNbChars()) : 4;
+        final byte[] flags = Arrays.copyOfRange(incomingMessage, idxFlags, idxFlags + model.getRespNbFlags());
+        if (logger.isTraceEnabled()) {
+            for (int i = 1; i <= flags.length; i++) {
+                try {
+                    logger.trace("handleValidHexMessage: Flag {} = {} bits 7-0 = {} {} {} {} {} {} {} {}", i,
+                            Integer.toHexString(flags[i - 1] & 0x000000FF), RotelFlagsMapping.isBitFlagOn(flags, i, 7),
+                            RotelFlagsMapping.isBitFlagOn(flags, i, 6), RotelFlagsMapping.isBitFlagOn(flags, i, 5),
+                            RotelFlagsMapping.isBitFlagOn(flags, i, 4), RotelFlagsMapping.isBitFlagOn(flags, i, 3),
+                            RotelFlagsMapping.isBitFlagOn(flags, i, 2), RotelFlagsMapping.isBitFlagOn(flags, i, 1),
+                            RotelFlagsMapping.isBitFlagOn(flags, i, 0));
+                } catch (RotelException e1) {
+                }
+            }
+        }
+        try {
+            dispatchKeyValue(KEY_POWER_ZONE2, model.isZone2On(flags) ? POWER_ON : STANDBY);
+        } catch (RotelException e1) {
+        }
```

Comment:
What's the reason all these catch blocks are empty? If this is intentional, you could add a comment
```suggestion
        try {
            dispatchKeyValue(KEY_POWER_ZONE2, model.isZone2On(flags) ? POWER_ON : STANDBY);
        } catch (RotelException e1) {
            // nothing
        }
```

Reply:
Thank you @fwolter for your review.
I checked all RotelException catch and added a comment when nothing was done.